src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -759,6 +770,17 @@ lemma prod_comparison_inv_natural (f : A ⟶ A') (g : B ⟶ B')
     prod.map (F.map f) (F.map g) ≫ inv (prod_comparison F A' B') :=
 by rw [is_iso.eq_comp_inv, category.assoc, is_iso.inv_comp_eq, prod_comparison_natural]
 
+/--
+The natural isomorphism `F(A ⨯ -) ≅ FA ⨯ F-`, provided each `prod_comparison F A B` is an
+isomorphism (as `B` changes).
+-/
+@[simps {rhs_md := semireducible}]
+def prod_comparison_nat_iso [has_binary_products C] [has_binary_products D]
```

Comment:
Should this be connected to some `preserves_products` assumption?

Reply:
It already is! Once `category_theory.limits.preserves.shapes.binary_products` is imported, and we have the assumption `preserves_limits_of_shape (discrete walking_pair) F` (which is the way of writing "preserves binary products"), we have `[∀ B, is_iso (prod_comparison F A B)]` as an instance.